src/components/CourseItemDialog.tsx

```diff
@@ -304,6 +315,63 @@ const CourseItemDialog: React.FC<CourseItemDialogProps> = ({
         };
     }, []);
 
+    // Add beforeunload event listener to prevent page reload/close with unsaved changes
+    useEffect(() => {
+        const handleBeforeUnload = (e: BeforeUnloadEvent) => {
+            const hasChanges = activeItem.type === 'material'
+                ? learningMaterialEditorRef.current?.hasChanges() || false
+                : quizEditorRef.current?.hasChanges() || false;
+            // Only show warning if user is in edit mode or if there are actual unsaved changes
+            if (hasChanges) {
+                e.preventDefault();
+                e.returnValue = 'You have unsaved changes. Are you sure you want to leave?';
+                return 'You have unsaved changes. Are you sure you want to leave?';
+            }
+        };
+
+        // Add the event listener
+        window.addEventListener('beforeunload', handleBeforeUnload);
+
+        // Clean up the event listener
+        return () => {
+            window.removeEventListener('beforeunload', handleBeforeUnload);
+        };
+    }, [activeItem, dialogTitleRef, isEditMode]);
+
+    // Handle browser back/forward navigation
+    useEffect(() => {
+        const handlePopState = (e: PopStateEvent) => {
+            const hasChanges = activeItem.type === 'material'
+                ? learningMaterialEditorRef.current?.hasChanges() || false
+                : quizEditorRef.current?.hasChanges() || false;
+            // Prevent navigation if user is in edit mode, the item is a draft, or if there are unsaved changes
+            if (isEditMode || activeItem?.status === 'draft' || hasChanges) {
```

Comment:
why is `||` being used? If there are no changes and user is in edit mode, why should we prevent?